src/client/components/Navigation/LanguageSettings.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import { Popover } from 'antd';
+import { connect } from 'react-redux';
+import { availableLocalesToReactIntl } from '../../translations/index';
+import { SUPPORTED_LANGUAGES } from '../../../common/constants/settings';
+import { saveSettings, setLocale } from '../../settings/settingsActions';
+import { getIsAuthenticated, getLocale } from '../../reducers';
+import PopoverMenu, { PopoverMenuItem } from '../PopoverMenu/PopoverMenu';
+import './LanguageSettings.less';
+
+@connect(
+  state => ({
+    authenticated: getIsAuthenticated(state),
+    locale: getLocale(state),
+  }),
+  {
+    saveSettings,
+    setLocale: locale => dispatch => dispatch(setLocale(locale)),
```

Comment:
AFAIK this would be the same as just writing `setLocale` instead of this whole line.

Reply:
yeah looks like it is, seems to be working fine when I changed it